recipes/ncbi-cxx-toolkit-public/all/conanfile.py

```diff
@@ -0,0 +1,294 @@
+
+        requirements = set()
+        for component in _required_components:
+            libraries = self._tk_dependencies["libraries"][component]
+            for lib in libraries:
+                if lib in self._tk_dependencies["requirements"].keys():
+                    requirements.update(self._tk_dependencies["requirements"][lib])
+
+        for req in requirements:
+            pkgs = self._translate_req(req)
+            if pkgs is not None:
+                for pkg in pkgs:
+                    print("Package requires ", pkg)
+                    self.requires(pkg)
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        if self.settings.os not in ["Linux", "Macos", "Windows"]:
+            raise ConanInvalidConfiguration("This operating system is not supported")
+        if is_msvc(self):
+            check_min_vs(self, 192)
+            if self.options.shared and is_msvc_static_runtime(self):
+                raise ConanInvalidConfiguration("This configuration is not supported")
+        else:
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(f"This version of {self.settings.compiler} is not supported")
+        if cross_building(self):
+            raise ConanInvalidConfiguration("Cross compilation is not supported")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        apply_conandata_patches(self)
+        root = os.path.join(os.getcwd(), "CMakeLists.txt")
+        with open(root, "w", encoding="utf-8") as f:
+            f.write("cmake_minimum_required(VERSION 3.15)\n")
+            f.write("project(ncbi-cpp)\n")
+            f.write("include(src/build-system/cmake/CMake.NCBItoolkit.cmake)\n")
+            f.write("add_subdirectory(src)\n")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["NCBI_PTBCFG_PACKAGING"] = True
+        if self.options.shared:
+            tc.variables["NCBI_PTBCFG_ALLOW_COMPOSITE"] = True
+        tc.variables["NCBI_PTBCFG_PROJECT_LIST"] = str(self.options.with_projects) + ";-app/netcache"
+        if self.options.with_targets != "":
+            tc.variables["NCBI_PTBCFG_PROJECT_TARGETS"] = self.options.with_targets
+        if len(self._componenttargets) != 0:
+            tc.variables["NCBI_PTBCFG_PROJECT_COMPONENTTARGETS"] = ";".join(self._componenttargets)
+        if is_msvc(self):
+            tc.variables["NCBI_PTBCFG_CONFIGURATION_TYPES"] = self.settings.build_type
+        tc.generate()
+        cmdep = CMakeDeps(self)
+        cmdep.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+# Visual Studio sometimes runs "out of heap space"
```

Comment:
Are there any links to this happening, either upstream of or on CCI so we can have a refernece to see if the problem is fixed?

Reply:
There is no links.
We struggled with the problem several years ago, when adopting CMake.
The workaround that finally helped was using devenv.com (instead of devenv.exe) to build the solution.
It works fine in CI. When building it in Visual Studio IDE, we still have this problem.
It is less painfull though. People do not often build the whole toolkit. Also, running the build second time, usually succeeds.
With Conan package, we saw this problem again. We do not want to try our luck. Disabling parallel build is a safe choice.